src/promql.grammar

```diff
@@ -254,9 +254,13 @@ AtModifierPreprocessors {
   whitespace { std.whitespace+ }
   LineComment { "#" ![\n]* }
 
-  NumberLiteral { // TODO: Handle NaN etc.
-    (std.digit+ ("." std.digit*)? | "." std.digit+) (("e" | "E") ("+" | "-")? std.digit+)? |
-    "0x" (std.digit | $[a-fA-F])+
+  NumberLiteral {
```

Comment:
looks good for me honestly. I'm just wondering if it couldn't make sense to have that instead:

```
SignedNumber{
  (Add | Sub ) NumberLiteral
}
```

Like that, we are able to know if it's classic number or a signed number and the sign of it as well.

It's just a proposal, but I'm feeling it's a bit cleaner to introduce this new rule.

And I think it should be ok with the precedence too.

Reply:
I'm talking about that also because like that we could avoid to introduce this new Rule just for the @ thing. (PR #27 )

Reply:
Hmm, I would prefer to stay as close as possible to the upstream parser, for which any number (signed or not) just ends up being a number literal, without anything wrapped around it. And for me, a "number" normally already includes the concept of being negative or positive unless it's explicitly named "unsigned"...

I guess in the upstream parser, the main reason why for the new `@` conditions they don't just use `number_literal` is because the `number_literal` rule can only produce positive `NumberLiteral` AST nodes, and they abuse the `unary_expr` rule to produce negative `NumberLiteral` AST nodes. I wonder whether that can be cleaned up upstream actually, or if there's a reason it doesn't work better with the yacc-based parser...

Reply:
alright :). I was suspected a bit the answer. But still insteresting to ask it anyway.

I will merge the PR :). Thanks for improvement !